python/ray/air/_internal/usage.py

```diff
@@ -0,0 +1,18 @@
+from typing import TYPE_CHECKING
+
+from ray._private.usage.usage_lib import TagKey, record_extra_usage_tag
+
+if TYPE_CHECKING:
+    from ray.train.trainer import BaseTrainer
+
+
+def tag_air_trainer(trainer: "BaseTrainer"):
+    from ray.train.trainer import BaseTrainer
+
+    assert isinstance(trainer, BaseTrainer)
```

Comment:
This assertion seems fine, but should we wrap the whole function in a `try`/`except` so that it doesn't cause real user failures?

Reply:
This is purely controlled internally - external user will not call `tag_air_trainer`. The assertion is just to flag any issue during development (for internal ray developer like us).
I don't think a try-catch is needed here. Or if people really don't like assertion, I can remove it. And let's just rely on eyeball checking the type hinting.

Reply:
actually, why not make this util more generic, like make it a decorator
```
@air_telemetry("trainer")
class XxxTrainer:
    ...
```
you can then use it to tag all the other objects, like
```
@air_telemetry("searcher")
class ASHASearecher:
   ...
```
that would allow you to avoid the BaseTrainer import problem as well.

Reply:
I like the idea. Depending on what we want to do. e.g. do we want to know how many people provide a custom implementation of our interface? I think those metrics are still valuable. 
if yes, only tagging our classes may not be enough. 
cc @matthewdeng on what do we want to achieve..

Reply:
hmm, that's a good point. we should probably track # of custom trainers.
so in that case, we need to decorate the ``BaseTrainer`` with this.
note that this decorator still doesn't know anything about Trainer or what not.
so the dependency is still one-way.

Reply:
RLlib has a list of all the internal Algorithm types.
does Train have such a list somewhere?

Reply:
> does Train have such a list somewhere?

Not right now. One pattern might be:

1. Add a decorator on base abstract classes.
    i. We can define where the call to log is made.
2. Have a whitelist of Ray-provided class names.

This would work for cases where we expect one entity (e.g. trainers, searchers, schedulers) but we'd need to figure out something different for usage for other concepts where there might be more than one (e.g. callbacks)

Reply:
Actually maybe the decorator is unnecessary... I would say start with a simple utility function and we can add a decorator later if we think it cleans things up.

Reply:
I am fine either way. but @matthewdeng , are you suggesting `tag_trainer` to be in /train/ folder and `tag_scheduler` etc to be in /tune/ folder? or are you fine with /air folder but hide the dependencies?

decorator or not, we can hide dependency by just importing dependency within the utility function/decorator function.

Reply:
Personally I would lean towards `train`, but given that this is private and can easily be moved later, consider this a non-blocking opinion!

Reply:
Consider non-blocking.